itests/test-jdbc/src/test/java/org/apache/hive/jdbc/TestStandaloneJdbc.java

```diff
@@ -0,0 +1,206 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hive.jdbc;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.hadoop.hive.metastore.utils.SecurityUtils;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TestStandaloneJdbc {
+  private static File workDir = new File(System.getProperty("java.io.tmpdir"), "test-it-standalone-jdbc");
+  private static ITAbstractContainer HS2;
+  private static ITAbstractContainer ZK_HS2;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    workDir.mkdirs();
+    HS2 = new ITHiveServer2(workDir);
+    ZK_HS2 = new ITZKHiveServer2(workDir);
+
+    HS2.start();
+    ZK_HS2.start();
+  }
+
+  @Test
+  public void testBinaryJdbc() throws Exception {
+    testMetaOp(HS2.getBaseJdbcUrl(), true);
+    testDataSourceOp(HS2.getBaseJdbcUrl());
+    testNegativeJdbc(HS2.getBaseJdbcUrl());
+  }
+
+  private void testMetaOp(String url, boolean kerberos) throws Exception {
+    try (Connection con = DriverManager.getConnection(url)) {
+      try (Statement stmt = con.createStatement()) {
+        String tableName = "testHiveDriverTable1";
+        stmt.execute("drop table if exists " + tableName);
+        stmt.execute("create table " + tableName + " (key string, value string)");
+        // show tables
+        String sql = "show tables '" + tableName + "'";
+        System.out.println("Running: " + sql);
+        ResultSet res = stmt.executeQuery(sql);
+        if (res.next()) {
+          System.out.println(res.getString(1));
+        }
+        // describe table
+        sql = "describe " + tableName;
+        System.out.println("Running: " + sql);
+        res = stmt.executeQuery(sql);
+        while (res.next()) {
+          System.out.println(res.getString(1) + "\t" + res.getString(2));
+        }
+
+        // "Client cannot authenticate via:[TOKEN, KERBEROS]" in running the Kerberized tez local mode
+        if (!kerberos) {
```

Comment:
It looks like two different test cases: one with Kerberos, and one without Kerberos.
Also it only has assertion if it is a not Kerberos test case. 
In case of kerberos == true, it tests nothing.

Reply:
yeah, this is different from the jdbc unit test, we are verifying:
1. packaged all necessary classes into the standalone jdbc jar;
2. if have the compatible issue
If there are no problems running this queries, e.g, no class def found or unexpected exception, then we can say this standalone jar can work as expected

Reply:
The non-Kerberos case is Tez job needed, otherwise we would see the "Client cannot authenticate via:[TOKEN, KERBEROS]" while submitting the Tez job, I'm not sure how to solve it.